File: backend/app/schemas/toast.py
from __future__ import annotations

import uuid
from datetime import datetime
from decimal import Decimal
from typing import Optional

from pydantic import BaseModel, field_validator


class ToastConnectRequest(BaseModel):
    location_id: uuid.UUID
    client_id: str
    client_secret: str
    toast_restaurant_guid: str
    historical_import_from: Optional[datetime] = None


class ToastConnectResponse(BaseModel):
    location_id: uuid.UUID
    toast_restaurant_guid: str
    is_active: bool
    historical_import_complete: bool
    last_synced_at: Optional[datetime]

    model_config = {"from_attributes": True}


class ToastSyncJobResponse(BaseModel):
    id: uuid.UUID
    location_id: uuid.UUID
    job_type: str
    status: str
    started_at: Optional[datetime]
    completed_at: Optional[datetime]
    date_from: Optional[datetime]
    date_to: Optional[datetime]
    orders_synced: int
    employees_synced: int
    time_entries_synced: int
    error_message: Optional[str]
    triggered_by: Optional[uuid.UUID]
    created_at: datetime

    model_config = {"from_attributes": True}


class ManualSyncRequest(BaseModel):
    location_id: uuid.UUID
    sync_type: str = "incremental"  # incremental | historical
    date_from: Optional[datetime] = None
    date_to: Optional[datetime] = None

File: frontend/src/types/toast.ts
export interface ToastSyncConfig {
  location_id: string;
  toast_restaurant_guid: string;
  is_active: boolean;
  historical_import_complete: boolean;
  last_synced_at: string | null;
  historical_status: "pending" | "running" | "complete" | "failed" | null;
  historical_started_at: string | null;
  historical_orders_synced: number | null;
  historical_error: string | null;
}

export type SyncJobStatus = "pending" | "running" | "complete" | "failed";
export type SyncJobType = "incremental" | "historical" | "manual";

export interface ToastSyncJob {
  id: string;
  location_id: string;
  job_type: SyncJobType;
  status: SyncJobStatus;
  started_at: string | null;
  completed_at: string | null;
  date_from: string | null;
  date_to: string | null;
  orders_synced: number;
  employees_synced: number;
  time_entries_synced: number;
  error_message: string | null;
  triggered_by: string | null;
  created_at: string;
}
